LogicLayer/Notifications/MailSender.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using GameBoard.LogicLayer.Configurations;
+using Microsoft.Extensions.Options;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+
+namespace GameBoard.LogicLayer.Notifications
+{
+    public class MailSender : IMailSender
+    {
+        private MailNotificationsConfiguration MailOptions { get; }
+
+        public MailSender(IOptions<MailNotificationsConfiguration> mailOptionsAccessor)
```

Comment:
This `Accessor` is confusing/meaningless/weird (pick one). I think `mailOptions` would be enough.

Reply:
On second thought, this argument and the corresponding property is in fact `MailNotificationsOptions`, not `MailOptions`.

Reply:
Done